ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-composite-answer/activityCompositeAnswer.component.ts

```diff
@@ -176,4 +211,14 @@ export class ActivityCompositeAnswer implements OnChanges {
     private showAsterisk(questionType: QuestionType): boolean {
         return !this.config.compositeRequiredFieldExceptions.includes(questionType);
     }
+
+    private resetEquationsRowIndexes(compositeChildQuestionBlocks: ActivityQuestionBlock<any>[][]): void {
+        compositeChildQuestionBlocks.forEach((row: ActivityQuestionBlock<any>[], rowIndex: number) => {
+            for (const questionBlock of row) {
+                if (!(questionBlock.generatesAnswers())) {
+                    (questionBlock as ActivityEquationQuestionBlock).compositeRowIndex = rowIndex;
```

Comment:
OK. You ALMOST got rid of ActivityEquationQuestionBlock!
Just move `compositeRowIndex` up to ActivityQuestionBlock, set it for all the blocks and you are done.

Reply:
Done